### PR TITLE
*: "--unsafe-overwrite-db" flag to support v2 migration with no previous v3 data

### PR DESCRIPTION
**Pull request by gyuho**
Fix https://github.com/coreos/etcd/issues/9480.

### CODE REVIEW

File: Documentation/upgrades/upgrade_3_2.md [hunk 1]
```diff
@@ -212,6 +212,8 @@ See [issue #6336](https://github.com/coreos/etcd/issues/6336) for more contexts.
 
 ### Server upgrade checklists
 
+**NOTE:** `v3.2.18` added [`etcd --unsafe-overwrite-db`](https://github.com/coreos/etcd/pull/9484) flag, to support [migration from v2 with no v3 data](https://github.com/coreos/etcd/issues/9480). etcd server panics if it tries to restore from existing snapshots but no v3 `ETCD_DATA_DIR/member/snap/db` file. This happens when the server had migrated from v2 with no previous v3 data. This is to prevent accidental v3 data loss (e.g. `db` file might have been moved). With `--unsafe-overwrite-db` enabled, etcd allows to create a fresh `db` file on reboot, when there are existing snapshots. To continue to use etcd without v3 data, keep `--unsafe-overwrite-db` enabled. Or write some test v3 keys (e.g. `ETCDCTL_API=3 etcdctl put foo bar`), and then restart etcd with `--unsafe-overwrite-db` disabled. Use this flag only for v2 migration. Otherwise, previous v3 data could be overwritten with existing snapshots. v4 will deprecate this flag.
```
Comment (gyuho) [inline Documentation/upgrades/upgrade_3_2.md]: @jlhawn Please let me know if this makes sense. Basically, if you aren't going to write any v3 keys for awhile, keep the `--unsafe-overwrite-db` flag. Or start with `--unsafe-overwrite-db` flag once, and write some test v3 keys. And restart with `--unsafe-overwrite-db` flag disabled. It's complicated, but think it's safer :0

Comment (jlhawn) [reply Documentation/upgrades/upgrade_3_2.md]: @gyuho that makes sense. Thanks!

**Comment by xiang90** [comment]
2.3.x should only be allowed to upgrade to 3.0 directly. Why do we need to change 3.3 branch?

**Comment by gyuho** [comment]
@xiang90 @jlhawn The extra snapshot recovery check `recoverSnapshotBackend` with consistent index was only introduced in v3.2 (https://github.com/coreos/etcd/commit/f6cd4d4f5b819a9af9ae8af35dc80b59a5f42477). Before v3.2, it just silently creates an empty `db` file, until user finds out it errors out upgrading to v3.2.

On second thought, this PR doesn't really fix anything. Just allow users to run unsafe operations.

@jlhawn I would just suggest writing some test v3 keys on v3.1 to v3.2 migration path,  to work around https://github.com/coreos/etcd/issues/9480. Instead, I will make it clear that post v3.1 migration should only happen when you have data in v3 storage.

I will update the docs next week.

**Comment by gyuho** [comment]
Documented here https://github.com/coreos/etcd/pull/9486.

**Pull request closed without merging**